channeldb/payments.go

```diff
@@ -0,0 +1,211 @@
+package channeldb
+
+import (
+	"github.com/roasbeef/btcutil"
+	"io"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/boltdb/bolt"
+	"encoding/binary"
+	"bytes"
+)
+
+var (
+	// invoiceBucket is the name of the bucket within the database that
+	// stores all data related to payments.
+	// Within the payments bucket, each invoice is keyed by its invoice ID
+	// which is a monotonically increasing uint64.
+	// BoltDB sequence feature is used for generating monotonically increasing
+	// id.
+	paymentBucket = []byte("payments")
+)
+
+type OutgoingPayment struct {
+	Invoice
+	// Total fee paid
+	Fee btcutil.Amount
+	// Path including starting and ending nodes
+	Path [][]byte
+	TimeLockLength uint64
```

Comment:
As a space savings measure, this can instead be `uint32` since all timelocks in Bitcoin are expressed using solely a 32-bit integer.